examples/napi/src/threadsafe_function.rs

```diff
@@ -3,61 +3,158 @@ use std::thread;
 use napi::{
   bindgen_prelude::*,
   threadsafe_function::{ErrorStrategy, ThreadsafeFunction, ThreadsafeFunctionCallMode},
-  JsBoolean,
+  JsBoolean, JsNumber, JsString, JsUndefined,
 };
 
-#[napi]
 pub fn call_threadsafe_function(callback: JsFunction) -> Result<()> {
   let tsfn: ThreadsafeFunction<u32, ErrorStrategy::CalleeHandled> = callback
-    .create_threadsafe_function(0, |ctx| {
-      ctx.env.create_uint32(ctx.value + 1).map(|v| vec![v])
-    })?;
-  for n in 0..100 {
+    .create_threadsafe_function(
+      0,
+      |ctx| ctx.env.create_uint32(ctx.value + 1).map(|v| vec![v]),
+      |res: Promise<u32>| {},
```

Comment:
`res: Promise<T>` is dropped immediately after promise get constructed, so the channel inside it is closed as well. 

```
// RUST, simplified
promise.then((v) => { 
  // `.then` callback will be getting called in next loop
  // so the cpu keeps going on the rust side, till here(commented line), where the created promise dropped
  sender.send(v)
})`
```

Reply:
Thank you for the quick note!